buildSrc/src/main/groovy/org/mockito/workflow/gradle/internal/ReleaseWorkflowExtension.java

```diff
@@ -0,0 +1,128 @@
+package org.mockito.workflow.gradle.internal;
+
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.specs.Spec;
+import org.mockito.workflow.gradle.ReleaseWorkflow;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+
+public class ReleaseWorkflowExtension implements ReleaseWorkflow {
+
+    final List<Task> steps = new ArrayList<Task>();
+    Task previousStep;
+    Task previousRollback;
+    final List<Task> rollbacks = new ArrayList<Task>();
+    private final Project project;
+    private final List<Callable<Boolean>> allowers = new ArrayList<Callable<Boolean>>();
+
+    public ReleaseWorkflowExtension(Project project) {
+        this.project = project;
+    }
+
+    public ReleaseWorkflowExtension step(Task task, Map<String, Task> config) {
+        addStep(task, config);
+        return this;
+    }
+
+    public ReleaseWorkflowExtension step(Task task) {
+        return step(task, Collections.<String, Task>emptyMap());
+    }
+
+    public ReleaseWorkflowExtension onlyIf(Callable<Boolean> predicate) {
+        allowers.add(predicate);
+        return this;
+    }
+
+    private void addStep(final Task task, Map<String, Task> config) {
+        //populate steps collection
+        steps.add(task);
+
+        //main release task will depend on this step
+        project.getTasks().getByName("release").dependsOn(task);
+
+        //release steps must be sequential
+        if (previousStep != null) {
+            if (project.hasProperty("singleStep")) {
+                //special case, release steps do not form any sequence
+                //allowing the user to decide on the sequence from command line
+            } else {
+                //default mode, release steps form a sequence using 'dependsOn'
+                //forcing execution of previous steps
+                task.dependsOn(previousStep);
+            }
+        }
+        previousStep = task;
+
+        //only run task if it was allowed
+        for (final Callable<Boolean> allower : allowers) {
+            task.onlyIf(new SpecAdapter(allower));
+        }
+
+        StepConfiguration stepConfig = new StepConfiguration(config);
+
+        if (stepConfig.isEmpty()) {
+            return; //no rollback/cleanup configured
+        }
+
+        Task rollback = stepConfig.getRollback();
+        if (rollback != null) {
```

Comment:
I'd move this if / else logic in `StepConfiguration` since this is this class that perform the check : https://github.com/mockito/mockito/pull/724/files#diff-ffbd6e3d52cfacae9bcc7e90cbd71e72R20